src/context/AppProvider/AppContext.tsx

```diff
@@ -134,10 +135,27 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
 
           const accountId = accountIds[idx]
           const { chainId } = fromAccountId(accountId)
+
           const { hasActivity } = account.accounts.byId[accountId]
 
+          const accountNumberHasChainActivity = !isUtxoChainId(chainId)
+            ? hasActivity
+            : // For UTXO AccountIds, we need to check if *any* of the scriptTypes have activity, not only the current one
+              // else, we might end up with partial account data, with only the first 1 or 2 out of 3 scriptTypes
+              // being upserted for BTC and LTC
+              accountResults.some((res, _idx) => {
+                if (res.status === 'rejected') return false
+                const { data: account } = res.value
+                if (!account) return false
+                const accountId = accountIds[_idx]
+                const { chainId: _chainId } = fromAccountId(accountId)
+                if (chainId !== _chainId) return false
+                return account.accounts.byId[accountId].hasActivity
+              })
```

Comment:
This looks relevant to `src/components/ManageAccountsDrawer/helpers.ts` -> `getAccountIdsWithActivityAndMetadata`.

We should centralize this so it doesnt get lost as we currently hav ea high rate of change with these queries

Reply:
@woodenfurniture would need to clarify intent of `getAccountIdsWithActivityAndMetadata` - as far as I can see those are two different concerns - this is intended to get account *number* activity while `getAccountIdsWithActivityAndMetadata` looks like it's supposed to get the granular activity per `AccountId`, at least that's what the current implementation suggests.

Regardless, we don't currently consume `hasActivity` from `getAccountIdsWithActivityAndMetadata` so unless that's something we plan to consume, we should probably rename it to `getAccountIdsWithMetadata` and remove the notion of `hasActivity` there?

Reply:
@woodenfurniture bumpy bump